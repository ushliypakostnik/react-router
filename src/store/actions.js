const fetchUrl = process.env.REACT_APP_API_URL;

export const pageToActive = (page) => ({
  type: "PAGE_TO_ACTIVE",
  activePage: page
});

export const setMinHeight = (minHeight) => ({
  type: "SET_MIN_HEIGHT",
  minHeight: minHeight
});

export const toogleTheme = (theme) => ({
  type: "TOGGLE_THEME",
  theme: theme
});

export const requestAlbums = () => ({
  type: "REQUEST_ALBUMS"
});

export const receiveAlbums = (albums) => ({
  type: "RECEIVE_ALBUMS",
  albums
});

export const requestAlbumFailed = (error) => ({
  type: "REQUEST_ALBUMS_FAILED",
  error
});

export const fetchAlbums = () => {
  return dispatch => {
    dispatch(requestAlbums());
    return fetch(fetchUrl + "/albums")
      .then(res => res.json())
      .then(
        (result) => {
          /* eslint-disable array-callback-return */
          let resultArr = [];
          result.map((item, index) => {
            if (index === 0) {
              resultArr.push({text: item.name, path: "/"});
            } else {
              resultArr.push({text: item.name, path: "/" + item.id});
            }
          });
          dispatch(receiveAlbums(resultArr));
        },
        (error) => {
          dispatch(requestAlbumFailed(error));
        }
      );
  }
}

export const requestData = () => ({
  type: "REQUEST_DATA"
});

export const receiveData = (data) => ({
  type: "RECEIVE_DATA",
  data
});

export const requestDataFailed = (error) => ({
  type: "REQUEST_DATA_FAILED",
  error
});

export const fetchData = (album) => {
  return dispatch => {
    dispatch(requestData());
    return fetch(fetchUrl + "/albums/" + album)
      .then(res => res.json())
      .then(
        (result) => {
          dispatch(receiveData(result));
        },
        (error) => {
          dispatch(requestDataFailed(error));
        }
      );
  }
}
