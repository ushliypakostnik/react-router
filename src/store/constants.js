export const FETCH_URL = process.env.REACT_APP_API_URL;

export const THEME = {
  DARK: "dark",
  LIGHT: "light"
};

export const COOKIES = {
  THEME: "theme"
};
