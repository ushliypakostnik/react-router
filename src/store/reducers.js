const reducer = (state = { }, action) => {
  switch (action.type) {
    case "REQUEST_ALBUMS":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_ALBUMS":
      return Object.assign({}, state, {
        isFetching: false,
        albums: action.albums
      });
    case "REQUEST_ALBUMS_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "REQUEST_DATA":
      return Object.assign({}, state, {
        isFetching: true
      });
    case "RECEIVE_DATA":
      return Object.assign({}, state, {
        isFetching: false,
        data: action.data
      });
    case "REQUEST_DATA_FAILED":
      return Object.assign({}, state, {
        isFetching: false,
        error: action.error
      });
    case "PAGE_TO_ACTIVE":
      return Object.assign({}, state, {
        activePage: action.activePage
      });
    case "TOGGLE_THEME":
      return Object.assign({}, state, {
        theme: action.theme
      });
    default:
      return state;
  }
}

export default reducer;
