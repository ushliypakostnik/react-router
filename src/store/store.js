import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunkMiddleware from 'redux-thunk';
import { createLogger } from 'redux-logger';
import { browserHistory } from "react-router";
//import { createBrowserHistory } from "history";
import { syncHistoryWithStore, routerReducer } from 'react-router-redux';

import { INITIAL_STATE } from './constants';
import { pageToActive } from './actions';
import reducer from './reducers';

const middlewares = [];
middlewares.push(thunkMiddleware)

if (process.env.NODE_ENV !== 'production') {
  const loggerMiddleware = createLogger();

  middlewares.push(loggerMiddleware);
}

function configureStore(state) {
  return createStore(
    combineReducers({
      reducer,
      routing: routerReducer
    }),
    state,
    applyMiddleware(...middlewares)
  );
}

const store = configureStore(INITIAL_STATE);

//const history = createBrowserHistory();
export const history = syncHistoryWithStore(browserHistory, store);
//history.listen(location => console.info('-> location:', location));

let lastUrl = "";
if (store.getState().routing.locationBeforeTransitions != null) {
  lastUrl = store.getState().routing.locationBeforeTransitions.pathname;
} else {
  lastUrl = "/";
}

//store.subscribe(() => {
//  console.log("Store: ", store.getState());
//  console.log("last Url: ", lastUrl);
//});

store.dispatch(pageToActive(lastUrl));

export default store;
