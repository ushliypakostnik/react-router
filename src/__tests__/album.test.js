import React from 'react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { FETCH_URL, INITIAL_STATE } from '../store/constants';

import Album from '../components/album';

describe('album render', () => {

  const mockStore = configureStore();
  let store,
      wrapper,
      gallery;

  const props = {
    photos: [
      {src: `${FETCH_URL}/media/pinhole/010.jpg`, width: 620, height: 632},
      {src: `${FETCH_URL}/media/pinhole/020.jpg`, width: 1568, height: 632},
      {src: `${FETCH_URL}/media/pinhole/030.jpg`, width: 1053, height: 632}
    ]
  };

  beforeEach(() => {
    store = mockStore(INITIAL_STATE);
    wrapper = mount(<Provider store={store}>
                      <Album photos={props.photos} >
                      </Album>
                    </Provider>);
  });

  it('album render correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('album has wrapper with class', () => {
    expect(wrapper.find('.app__gallery')).toHaveLength(1);
  });

  it('album has gallery', () => {
    expect(wrapper.find('Gallery')).toHaveLength(1);
  });

  afterEach(() => wrapper.unmount());
});
