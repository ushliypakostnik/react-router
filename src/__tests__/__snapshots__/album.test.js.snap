// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`album render album render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Album)
    photos={
      Array [
        Object {
          "height": 632,
          "src": "http://127.0.0.1:8082/media/pinhole/010.jpg",
          "width": 620,
        },
        Object {
          "height": 632,
          "src": "http://127.0.0.1:8082/media/pinhole/020.jpg",
          "width": 1568,
        },
        Object {
          "height": 632,
          "src": "http://127.0.0.1:8082/media/pinhole/030.jpg",
          "width": 1053,
        },
      ]
    }
  >
    <Album
      dispatch={[Function]}
      photos={
        Array [
          Object {
            "height": 632,
            "src": "http://127.0.0.1:8082/media/pinhole/010.jpg",
            "width": 620,
          },
          Object {
            "height": 632,
            "src": "http://127.0.0.1:8082/media/pinhole/020.jpg",
            "width": 1568,
          },
          Object {
            "height": 632,
            "src": "http://127.0.0.1:8082/media/pinhole/030.jpg",
            "width": 1053,
          },
        ]
      }
    >
      <div
        className="app__gallery"
      >
        <Gallery
          direction="row"
          margin={0}
          onClick={[Function]}
          photos={
            Array [
              Object {
                "height": 632,
                "src": "http://127.0.0.1:8082/media/pinhole/010.jpg",
                "width": 620,
              },
              Object {
                "height": 632,
                "src": "http://127.0.0.1:8082/media/pinhole/020.jpg",
                "width": 1568,
              },
              Object {
                "height": 632,
                "src": "http://127.0.0.1:8082/media/pinhole/030.jpg",
                "width": 1053,
              },
            ]
          }
          targetRowHeight={350}
        >
          <div>
            Â 
          </div>
        </Gallery>
      </div>
    </Album>
  </Connect(Album)>
</Provider>
`;
